js/core/modules_registry.js

```diff
@@ -0,0 +1,21 @@
+const registry = {};
+
+export function setModule(name, module) {
+    registry[name] = module;
+}
+export function getModule(name) {
+    return registry[name];
+}
+
+export function componentGetter(path) {
```

Comment:
I suggest you rename the function so that it would describe an action.

```suggestion
export function getComponent(path) {
```

```suggestion
export function queryComponent(path) {
```

Or in the following way:
```suggestion
export function componentFactory(path) {
```

Also, consider replacing `component` with `module`

Reply:
I like `componentFactory`.